src/index.js

```diff
@@ -520,6 +553,206 @@ class EventsCoreAPI {
+   * @param {*} digiSignature2 - I/O Events generated digital signature2
+   * @param {*} pubKeyUrl1 - Cloud Front public key url1
+   * @param {*} pubKeyUrl2 - Cloud Front public key url2
+   * @param {*} recipientClientId - target recipient client id
+   * @param {*} signedPayload - I/O Events proper signed payload
+   * @returns {boolean} true if either signatures are valid or false 
+   */
+  async __verifyDigitalSignature (digiSignature1, digiSignature2, pubKeyUrl1, pubKeyUrl2, recipientClientId, signedPayload) {
+    // fetch the encoded public keys from the shared package default params
+    let pubKey1Pem, pubKey2Pem
+    try {
+      const state = await stateLib.init()
+      pubKey1Pem = await this.__fetchFromCacheOrApi(pubKeyUrl1, state)
+      pubKey2Pem = await this.__fetchFromCacheOrApi(pubKeyUrl2, state)
+    } catch (error) {
+      logger.error('error occurred while fetching from the public key url %s or %s for client %s due to %s',
+        pubKeyUrl1, pubKeyUrl2, recipientClientId, error.message)
+      return this.__genErrorResponse(500, 'Error occurred while fetching Public Key')
+    }
+    // verify the digital signatures using public keys fetched from params
+    return this.__verifySignature(digiSignature1, digiSignature2, signedPayload, pubKey1Pem, pubKey2Pem, recipientClientId)
+  }
+
+  /**
+   * Wrapper to fetch the key from aio-lib-state, if not present fetch using
+   * the cloud front url and set in aio-lib-state 
+   * 
+   * @param {*} pubKeyUrl cloud front url of format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+   * @param {*} state aio-lib-state client
+   * @returns public key
+   */
+  async __fetchFromCacheOrApi (pubKeyUrl, state) {
+    let publicKey, publicKeyFileName
+    try {
+      // public key url is the cloud front url in this format https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+      publicKeyFileName = pubKeyUrl.substring(pubKeyUrl.lastIndexOf('/') + 1)
+      const res = await state.get(publicKeyFileName)
+      publicKey = res.value
+      logger.info('public key %s fetched from aio lib state', publicKey)
+    } catch (error) {
+      logger.error('error fetching from aio lib state due to => %s', error.message)
+      throw error
+    }
+
+    if (publicKey) {
+      return publicKey
+    }
+
+    logger.info('public key for url %s not present in state lib cache, fetching from api..', pubKeyUrl)
+    // fetch from api and set in cache default expiry 24h
+    publicKey = this.__fetchPublicKey(pubKeyUrl)
+    await state.put(publicKeyFileName, { anObject: publicKey })
+    logger.info('fetched key set in the aio lib state')
+    return publicKey
+  }
+
+  /**
+   * Fetches public key using the cloud front public key url
+   * 
+   * @param {*} publicKeyUrl - cloud front public key url of format 
+   * https://d2wbnl47xxxxx.cloudfront.net/pub-key-1.pem
+   * @returns public key
+   */
+  __fetchPublicKey (publicKeyUrl) {
```

Comment:
Can we call this method `fetchPublicKeyFromCloudFront`?

Reply:
okay